spectrocloud/data_source_application_profile.go

```diff
@@ -29,12 +33,24 @@ func dataSourceApplicationProfileRead(_ context.Context, d *schema.ResourceData,
 	c := m.(*client.V1Client)
 	var diags diag.Diagnostics
 	if name, okName := d.GetOk("name"); okName {
-		application_profile, err := c.GetApplicationProfileByName(name.(string))
-		if err != nil {
-			return diag.FromErr(err)
+		if version, okVersion := d.GetOk("version"); okVersion {
+			applicationProfile, appUID, getVersion, err := c.GetApplicationProfileByNameAndVersion(name.(string), version.(string))
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			d.SetId(appUID)
+			d.Set("name", applicationProfile.Metadata.Name)
+			d.Set("version", getVersion)
+		} else {
+			applicationProfile, err := c.GetApplicationProfileByName(name.(string))
```

Comment:
I suggest using one function and passing there a default value 1.0.0 as version. This will be consistent with cluster profiles.

Reply:
Usually by default API will return the UID of the updated version of the application profile.  Hence if we set 1.0.0 as the default version user will always use the older version. Hence handled it separately.  will try to handle it in single function